### PR TITLE
Separate Replication2 and Graph test directory with their own CMakeLists.txt

### PR DESCRIPTION
**Pull request by markuspf**
### Scope & Purpose

This PR changes how the (previously already existing) mechanisms for building separate binaries for Replication2 and Graph tests work: 
 * It creates an object library of shared test code 
 * It creates object libraries of test cases for Replication2 and Graph
 * It uses the above library to provide all tests in the monolithic `arangodbtests` executable *as well as* the optionally built `arangodbtests_replication2` and  `arangodbtests_graph` 
 
It also demonstrates how easy it could be to split the monolithic, slow compiling (linking) arangodbtests binary into smaller entities, encouraging better separation, quicker testing, and developer happiness.

*BIG FAT WARNING* there is currently a massive footgun hidden in how tests are put together. If you use `add_library(foo STATIC ...)` your chippy linker will remove your googletests and they won't be run.

if you dont add your object library to the `arangodbtests` executable, your tests won't be run.

I hope to address this by running the test code with coverage checks to ascertain that tests are run.

Enterprise friend: https://github.com/arangodb/enterprise/pull/1129

### CODE REVIEW

File: tests/Graph/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    arango_tests_basics
+    velocypack
+    fmt)
+
+add_executable(arangodbtests_graph
+  EXCLUDE_FROM_ALL)
+
+target_link_libraries(arangodbtests_graph
+    gtest
+    arango_tests_basics
```
[the remark this answers is not in the document]
Comment (jvolmer) [reply tests/Graph/CMakeLists.txt:40]: There are more of these cases, e.g. in `target_link_libraries(arangodbtests ...)`.

**Review by jvolmer — COMMENTED** [review]
I really like that! I just have some comments where you could get rid of linking or including directories because this is already "inherited" from another library you already linked.

File: tests/Graph/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  SingleServerProviderTest.cpp
+  ${ADDITIONAL_GRAPH_TEST_SOURCES})
+
+target_include_directories(arango_tests_graph PUBLIC
+    ${PROJECT_SOURCE_DIR}/arangod
+    ${PROJECT_SOURCE_DIR}/lib
+    ${PROJECT_SOURCE_DIR}/${ENTERPRISE_INCLUDE_DIR}
+    ${PROJECT_SOURCE_DIR}/tests
+    ${PROJECT_SOURCE_DIR}/tests/Mocks/
+  )
```
[the remark this answers is not in the document]
Comment (markuspf) [reply tests/Graph/CMakeLists.txt]: Yes, good idea, thanks!

File: tests/Graph/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (markuspf) [reply tests/Graph/CMakeLists.txt:32]: Yes, I think I was too generous throwing link libraries around; I am not sure whether i really want to make linking of libraries explicit; I would rather that no *superflous* dependencies are given ;)

I'll have another go over the CMakeLists

**Review by jvolmer — APPROVED** [review]

**Pull request merged**